src/core/Akka.FSharp/FaultHandling.fs

```diff
@@ -0,0 +1,135 @@
+
+
+[<AutoOpen>]
+module Logging = 
+    open Akka.Event
+    
+    /// Logs a message using configured Akka logger.
+    let log (level : LogLevel) (mailbox : Actor<'Message>) (msg : string) : unit = 
+        let logger = mailbox.Log.Force()
+        logger.Log(level, msg)
+    
+    /// Logs a message at Debug level using configured Akka logger.
+    let inline logDebug mailbox msg = log LogLevel.DebugLevel mailbox msg
+    
+    /// Logs a message at Info level using configured Akka logger.
+    let inline logInfo mailbox msg = log LogLevel.InfoLevel mailbox msg
+    
+    /// Logs a message at Warning level using configured Akka logger.
+    let inline logWarning mailbox msg = log LogLevel.WarningLevel mailbox msg
+    
+    /// Logs a message at Error level using configured Akka logger. 
+    let inline logError mailbox msg = log LogLevel.ErrorLevel mailbox msg
+    
+    /// Logs an exception message at Error level using configured Akka logger.
+    let inline logException mailbox (e : exn) = log LogLevel.ErrorLevel mailbox (e.Message)
+    
+    open Printf
+    
+    let inline private doLogf level (mailbox : Actor<'Message>) msg = mailbox.Log.Value.Log(level, msg) |> ignore
+    
+    /// Logs a message using configured Akka logger.
+    let inline logf (level : LogLevel) (mailbox : Actor<'Message>) = kprintf (doLogf level mailbox)
+    
+    /// Logs a message at Debug level using configured Akka logger.
+    let inline logDebugf mailbox = kprintf (doLogf LogLevel.DebugLevel mailbox)
+    
+    /// Logs a message at Info level using configured Akka logger.
+    let inline logInfof mailbox = kprintf (doLogf LogLevel.InfoLevel mailbox)
+    
+    /// Logs a message at Warning level using configured Akka logger.
+    let inline logWarningf mailbox = kprintf (doLogf LogLevel.WarningLevel mailbox)
+    
+    /// Logs a message at Error level using configured Akka logger. 
+    let inline logErrorf mailbox = kprintf (doLogf LogLevel.ErrorLevel mailbox)
+
+open Akka.Util
+
+type ExprDeciderSurrogate(serializedExpr : byte array) = 
+    member __.SerializedExpr = serializedExpr
+    interface ISurrogate with
+        member this.FromSurrogate _ = 
+            let fsp = Nessos.FsPickler.FsPickler.CreateBinary()
+            let expr = 
+                (Serialization.deserializeFromBinary fsp (this.SerializedExpr) typeof<Expr<exn -> Directive>>) :?> Expr<exn -> Directive>
+            ExprDecider(expr) :> ISurrogated
+
+and ExprDecider(expr : Expr<exn -> Directive>) = 
+    member __.Expr = expr
+    member private this.Compiled = lazy this.Expr.Compile () ()
+    
+    interface IDecider with
+        member this.Decide(e : exn) : Directive = this.Compiled.Value(e)
+    
+    interface ISurrogated with
+        member this.ToSurrogate _ = 
+            let fsp = Nessos.FsPickler.FsPickler.CreateBinary()
+            ExprDeciderSurrogate(Serialization.serializeToBinary fsp this.Expr) :> ISurrogate
+
+type Strategy = 
```

Comment:
I took a look back at 85545c3cc1159c48acab9a35f615f070b82f2d60, which is where I believe the current shape came from. While I definitely agree that the previous shape was wrong, I'm not sure this shape is quite right either. This effectively becomes a thin wrapper around the C# API, and I think we can expose it in a more F#-friendly way. I'll push up a changeset shortly that incorporates what i was thinking.